### PR TITLE
[MIA-170,168,251,263, and 249]: Code Review Contribs

### PR DESCRIPTION
**Pull request by scottqueen-bixal**
needs to be considered and merged before https://github.com/GSA/made-in-america-data/tree/stage can be merged into https://github.com/GSA/made-in-america-data/tree/develop

Description
after a review of the code on `stage` branch, the following issues were apparent and this code provides solutions for them.

issues included:
- eslint engine not included
- unused static methods
- zombie code
- lint ignore rules with page-level overrides does not include a comment
- declarations not being used
- unused files existed in tree
- eslint not configured for mocha or chai
- test directory was being lint ignored
- some code unnecessarily eslint ignored
- variable casing not following best practices or consistent

How to test
- ensure code is sync'd with latest from `stage`
- review code
- execute ` node buildwaiverdata.js`
- execute ` node buildcsvdata.js`
- execute ` npm run test`

To prepare for code review, please follow the following checks.

 - [x] branch has detailed description of work/updates
 - [ ] title format follows: [Ticket Number] : Brief statement describing what this pull request solves.
 - [x] passed all integrated testing
 - [ ] pipeline has completed successfully
 - [ ] code review/design review requested as appropriate
 - [ ] pipeline link updated in ticket

### CODE REVIEW

**Comment by clmedders** [comment]
had to do a few extra steps, including here for testing reference 
- ensure code is sync'd with latest from stage
- review code
- ensure you are on the correct version of node, reference `.nvmrc`
- npm install 
- execute node buildwaiverdata.js
- execute node buildcsvdata.js
- execute npm run test

executed all three test steps twice, both were successful

**Review by clmedders — APPROVED** [review]

**Review by kennethedwards-bixal — COMMENTED** [review]
This is a smoke test that I want to keep. At the next PR I'll rename it to smokeTest for clarity.

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — COMMENTED** [review]
Will rename this to smokeTest at the next PR.

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

File: buildwaiverdata.js [hunk 1]
```diff
@@ -18,46 +20,41 @@ class DataScript {
     console.log('initiate')
   }
 
-  static add(a, b) {
```
Comment (scottqueen-bixal) [inline buildwaiverdata.js:21]: @kennethedwards-bixal is this the smoke test? feel free to propose a code edit here and in the spec and we can resolve the naming within this work

Comment (scottqueen-bixal) [reply buildwaiverdata.js:21]: you can simply include the suggestion in the comment then we can work it in as a commit on the fly
Screen Shot 2022-02-25 at 2 23 05 PM

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Review by kennethedwards-bixal — APPROVED** [review]

**Pull request merged**